Automattic/jetpack
# Resolves #37695 - Use clsx instead of classnames

Resolves #37695 - Use clsx instead of classnames

## Proposed changes:


`clsx` is [faster](https://github.com/lukeed/clsx/tree/925494cf31bcd97d3337aacd34e659e80cae7fe2/bench) and smaller  than `classnames`. Also, it is now used in Calypso (Automattic/wp-calypso#91408), in Gutenberg (WordPress/gutenberg#61138), and in `@wordpress` packages.

For consistency, this PR matches those changes:

* a rule was added to prevent future use of `classnames`: 6b5518d1653f1fcf4664b852c2b1cf6cad89cfa5
* `classnames` was replaced with `clsx`: b45ef169f05041337f0a52773237b3eb428febc0
* `classnames/dedupe` does not have an equivalent (lukeed/clsx#13), so its one use was refactored using `@wordpress/token-list`: a060f549765f07d892e4548165f8f650d6e614dd

Caveats:
* `jetpack-mu-wpcom` has [its own implementation](https://github.com/Automattic/jetpack/blob/fix/37695/classnames_to_clsx/projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/src/utils.ts#L9) of `classNames`; I've left that intact for now. If desired, we can take a look in a separate PR.
* `@automattic/social-previews` is a dependency, and the latest published version ([`2.0.1-beta.13`](https://www.npmjs.com/package/@automattic/social-previews)) still has a `classnames` dependency. This was removed in the aforementioned Calypso PR, but as of yet has not been published.

Internal reference: p4TIVU-aYc-p2

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Check the code for any typos or mistakes. Pay particularly close attention to the dedupe refactor.
